### PR TITLE
[MU4] notation_playback_layer_p1

### PR DESCRIPTION
**Pull request by vpereverzev**

### CODE REVIEW

File: src/framework/mpe/mpetypes.h [hunk 1]
```diff
@@ -476,6 +504,15 @@ struct ArticulationMeta
     duration_t overallDuration = 0;
     pitch_level_t overallPitchChangesRange = 0;
     dynamic_level_t overallDynamicChangesRange = 0;
+
+    bool operator==(const ArticulationMeta& other) const
+    {
+        return type == other.type
```
Comment (RomanPudashkin) [inline src/framework/mpe/mpetypes.h:510]: What about timestamp? Should we add it to this comparison?

Comment (vpereverzev) [reply src/framework/mpe/mpetypes.h:510]: sure, thanks

File: src/engraving/playback/playbackmodel.cpp [hunk 2]
```diff
@@ -87,31 +94,84 @@ void PlaybackModel::load(Ms::Score* score, async::Channel<int, int, int, int> no
...
+    const PlaybackEventList& eventsFromTick = trackPlaybackData->second.originData.at(timestamp);
+
+    PlaybackEventsMap result;
+
+    if (item->isChord()) {
+        for (const Ms::Note* note : Ms::toChord(item)->notes()) {
+            findEventsForNote(note, eventsFromTick, result[timestamp]);
+        }
+    } else {
+        findEventsForNote(Ms::toNote(item), eventsFromTick, result[timestamp]);
```
Comment (RomanPudashkin) [inline src/engraving/playback/playbackmodel.cpp:144]: I would suggest adding a check here to see if the item is really a note

**Review by RomanPudashkin — APPROVED** [review]

File: src/engraving/playback/playbackmodel.cpp (hunk 2 above)
Comment (vpereverzev) [reply src/engraving/playback/playbackmodel.cpp:144]: item->isPlayable() from above will ensure this

**Pull request merged**